src/tasks/acceptability.py

```diff
@@ -0,0 +1,433 @@
+"""Define the tasks and code for loading their data.
+
+- As much as possible, following the existing task hierarchy structure.
+- When inheriting, be sure to write and call load_data.
+- Set all text data as an attribute, task.sentences (List[List[str]])
+- Each task's val_metric should be name_metric, where metric is returned by
+get_metrics(): e.g. if task.val_metric = task_name + "_accuracy", then
+task.get_metrics() should return {"accuracy": accuracy_val, ... }
+"""
+import codecs
+import collections
+import copy
+import logging as log
+import os
+
+import numpy as np
+import torch
+
+from allennlp.training.metrics import CategoricalAccuracy
+from ..allennlp_mods.correlation import Correlation
+from allennlp.data import vocabulary
+from .registry import register_task
+
+# Fields for instance processing
+from allennlp.data import Instance, Token
+from allennlp.data.fields import TextField, LabelField, MultiLabelField, MetadataField, IndexField
+
+from ..utils.data_loaders import load_tsv, get_tag_list, BERT_MASK_TOK
+from .tasks import Task, sentence_to_text_field, SingleClassificationTask
+from .tasks import create_subset_scorers, update_subset_scorers, collect_subset_scores
+
+from typing import Iterable, Sequence, List, Dict, Any, Type
+
+
+@register_task("npi_pair_frozen", rel_path="NPI")
+class NPIMinimalPairFrozenTask(Task):
+    """ Task class for frozen minimal pair (cloze) acceptability judgement """
```

Comment:
How about 'NPIMinimalPairMLMTask' - MLM vs. classifier seems more important than frozen vs. tuned.

Reply:
Yes, that's good, I'll change this. This naming is pretty bad, especially so when considering that "frozen" is also used as a keyword for transfer paradigm.